src/core/ContentWindowController.cpp

```diff
@@ -198,34 +198,43 @@ void ContentWindowController::moveTo( const QPointF& position,
     default:
         return;
     }
-    constrainPosition( coordinates );
+    constrainPosition_( coordinates );
 
     contentWindow_.setCoordinates( coordinates );
 }
 
-void ContentWindowController::constrainSize( QSizeF& windowSize ) const
+void ContentWindowController::constrainSize_( QSizeF& windowSize ) const
 {
-    const QSizeF max_size = MAX_SIZE * displayGroup_.getCoordinates().size();
-    const QSizeF min_size = MIN_SIZE * displayGroup_.getCoordinates().size();
-
-    if ( windowSize.width() > max_size.width() ||
-         windowSize.height() > max_size.height( ))
-        windowSize.scale( max_size, Qt::KeepAspectRatio );
+    QSizeF maxSize = contentWindow_.getContent()->getDimensions();
```

Comment:
I think this does not apply for pdfs and svg: they typically have a default size, but could be resized by a much larger factor than just 2 since they can be rendered at an arbitrary resolution.